android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -79,24 +79,26 @@ private boolean isReactApplication(Context context) {
         return false;
     }
 
-    private void loadBundleLegacy() {
-        Activity currentActivity = getCurrentActivity();
-        Intent intent = currentActivity.getIntent();
-        currentActivity.finish();
-        currentActivity.startActivity(intent);
-
+    private void loadBundleLegacy(final Activity currentActivity) {
         mCodePush.invalidateCurrentInstance();
+
+        currentActivity.runOnUiThread(new Runnable() {
```

Comment:
I don't think it will actually help safe-guard us here, since the `loadBundle` method doesn't check against `null`, and therefore, would crash far before we hit this part, and the code which recreates the RN context couldn't `null` out the `Activity` reference we have, or force it to be GC'd, since we'd still be keeping it alive. The issue here was that we allowed RN to control the lifetime of the `Activity` reference, by delegating to `getCurrentActivity`, as opposed to capturing our own reference for the span of time that we need it.